otherlibs/stdune/src/io.ml

```diff
@@ -254,32 +254,92 @@ struct
        it fails for w/e reason *)
     external copyfile : string -> string -> unit = "stdune_copyfile"
 
-    external available : unit -> bool = "stdune_is_darwin"
+    external sendfile :
+      src:Unix.file_descr -> dst:Unix.file_descr -> int -> unit
+      = "stdune_sendfile"
+
+    external darwin : unit -> bool = "stdune_is_darwin"
+
+    external linux : unit -> bool = "stdune_is_linux"
+
+    let available =
+      if darwin () then `Copyfile else if linux () then `Sendfile else `Nothing
+
+    let sendfile =
+      let setup_copy ?(chmod = Fun.id) ~src ~dst () =
+        match Unix.openfile (Path.to_string src) [ O_RDONLY ] 0 with
+        | exception Unix.Unix_error (Unix.ENOENT, _, _) -> Error `Src_missing
+        | ic ->
```

Comment:
Why do you call this `ic`? Isn't this a file descriptor? `src_fd` seems like a clearer name. Same for `oc`.

Reply:
Will rename